vars/commonPipelineEnvironment.groovy

```diff
@@ -1,144 +1,28 @@
 import com.sap.piper.ConfigurationLoader
 import com.sap.piper.ConfigurationMerger
+import com.sap.piper.CommonPipelineEnvironment
 import com.sap.piper.analytics.InfluxData
 
 class commonPipelineEnvironment implements Serializable {
 
-    //stores version of the artifact which is build during pipeline run
-    def artifactVersion
+    // We forward everything to the singleton instance of
+    // commonPipelineEnvironment (CPE) on default value cache.
+    //
+    // Some background: each step has its own instance of CPE step.
+    // In case each instance has its own set of properties these instances
+    // are configured individually. Properties set on one instance cannot be
+    // retrieved with another instance. Now each instance forwards to one singleton.
+    // This means: all instances of the CPE shares the same properties/configuration.
 
-    //Stores the current buildResult
-    String buildResult = 'SUCCESS'
-
-    //stores the gitCommitId as well as additional git information for the build during pipeline run
-    String gitCommitId
-    String gitCommitMessage
-    String gitSshUrl
-    String gitHttpsUrl
-    String gitBranch
-
-    //GiutHub specific information
-    String githubOrg
-    String githubRepo
-
-    //stores properties for a pipeline which build an artifact and then bundles it into a container
-    private Map appContainerProperties = [:]
-
-    Map configuration = [:]
-    Map defaultConfiguration = [:]
-
-    String mtarFilePath
-    private Map valueMap = [:]
-
-    void setValue(String property, value) {
-        valueMap[property] = value
-    }
-
-    def getValue(String property) {
-        return valueMap.get(property)
-    }
-
-    String changeDocumentId
-
-    def reset() {
-        appContainerProperties = [:]
-        artifactVersion = null
-
-        configuration = [:]
-
-        gitCommitId = null
-        gitCommitMessage = null
-        gitSshUrl = null
-        gitHttpsUrl = null
-        gitBranch = null
-
-        githubOrg = null
-        githubRepo = null
-
-        mtarFilePath = null
-        valueMap = [:]
-
-        changeDocumentId = null
-
-        InfluxData.reset()
-    }
-
-    def setAppContainerProperty(property, value) {
-        appContainerProperties[property] = value
+    def methodMissing(String name, def args) {
```

Comment:
Do we really what to add meta programming or is that just a compatibility solution

Reply:
We forward everything to the new singleton instance. From my point of view this is not an abuse. Of course it is possible to write it down method by method. Lot of boilerplate code IMO.

Reply:
Please check if you are able to provide a "approve" like it is now. In case it is more easy for you to provide the "approach" with having the long version (forward each method individually) I can change it of course. IMO not an improvement, but whenever several humans act together we need to be able to make compromisses.